src/global/helpers/native-date-adapter.ts

```diff
@@ -0,0 +1,224 @@
+import { documentLanguage } from './language';
+
+export class NativeDateAdapter {
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  public static readonly DAYS_PER_WEEK = 7;
+
+  /**
+   * Creates an array with the given length and fills it by mapping with the provided function.
+   * @param length The length of the array to be created.
+   * @param valueFunction The function of array's index used to fill the array.
+   */
+  private _range<T>(length: number, valueFunction: (index: number) => T): T[] {
+    const valuesArray = Array(length);
+    for (let i = 0; i < length; i++) {
+      valuesArray[i] = valueFunction(i);
+    }
+    return valuesArray;
+  }
+
+  /** Creates a date but allows the month and date to overflow. */
+  private _createDateWithOverflow(year: number, month: number, date: number): Date {
+    const result = new Date(year, month, date);
+    // We need to correct for the fact that JS native Date treats years in range [0, 99] as
+    // abbreviations for 19xx.
+    if (year >= 0 && year < 100) {
+      result.setFullYear(this.getYear(result) - 1900);
+    }
+    return result;
+  }
+
+  /**
+   * Calculates the day of the week of the first day of the month, and then its offset from the first day of the week.
+   */
+  public getFirstWeekOffset(year: number, month: number): number {
+    const firstOfMonth = this.createDate(year, month, 1);
+    return (
+      (NativeDateAdapter.DAYS_PER_WEEK +
+        this.getDayOfWeek(firstOfMonth) -
+        this.getFirstDayOfWeek()) %
+      NativeDateAdapter.DAYS_PER_WEEK
+    );
+  }
+
+  /** Gets the year of the input date. */
+  public getYear(date: Date): number {
+    return date.getFullYear();
+  }
+
+  /** Gets the month of the input date. */
+  public getMonth(date: Date): number {
+    return date.getMonth();
+  }
+
+  /** Gets the day of the input date. */
+  public getDate(date: Date): number {
+    return date.getDate();
+  }
+
+  /** Gets the day of the week of the input date. */
+  public getDayOfWeek(date: Date): number {
+    return date.getDay();
+  }
```

Comment:
They were inherited from the angular version of the native date adapter, but since we're not using a "DateAdapter" interface in this case, and we're just dealing with native Date objects, I think we can get rid of them.